### DISCUSSION TITLE
Testing useJsApiLoader using react testing library

### DISCUSSION THREAD

**Discussion by Rafael805** [JustFly1984/react-google-maps-api#3047] [Q&A] [answered]
I'm trying the following but it doesn't work.

```
   jest.mock("@react-google-maps/api", () => ({
      useJsApiLoader: () => ({
        isLoaded: true,
        loadError: undefined,
      }),
    }));
```

**Comment by JustFly1984** [comment, marked as answer]
You should not use components and hooks from this package for jest unit tests.

**Reply by JustFly1984** [reply]
You can still use something like cypress to test user interactions, etc.
unit tests are useless in our case.

**Reply by Rafael805** [reply]
What if I have something like the following:

```
const { isLoaded } = useJsApiLoader({
    googleMapsApiKey: process.env.REACT_APP_GOOGLE_MAPS_API_KEY ?? "",
    libraries: ["places"],
  });

return (
  <>
   {isLoaded ?
      <SomeComponents />
     : 
       <LoadingSpinner />
    }
  </>
)
```

I then cannot test my component as it will always return `false`

**Reply by Rafael805** [reply]
Bump

**Reply by boroth** [reply]
Any followup here? I have code needs to be hidden/shown based on whether google maps API is loaded.

**Comment by boroth** [comment]
Is there a way to re-open this question? There was a follow-up but it wasn't answered.

**Reply by JustFly1984** [reply]
I've explained it a hundred times before, but how you gonna test injecting script into the DOM without browser? Answer is - You don't!

use e2e tests if you want. It is impossible to write unit tests for something what is not unit, nor testable.

**Reply by boroth** [reply]
I appreciate the condescension, but I'm not trying to get it to load in a unit test. I'm trying to get it to work with the google-maps approved mocks so that it at least doesn't blow up with errors because this library isn't handling checking for whether or not the JS API is actually loaded correctly.
